packages/react/src/QuestionnaireForm/QuestionnaireForm.tsx

```diff
@@ -41,9 +41,24 @@ export function QuestionnaireForm(props: QuestionnaireFormProps): JSX.Element |
   onSubmitRef.current = props.onSubmit;
 
   useEffect(() => {
-    setResponse(questionnaire ? buildInitialResponse(questionnaire) : undefined);
+    setResponse((prevResponse) => {
+      if (prevResponse) {
+        return prevResponse;
+      }
+      return questionnaire ? buildInitialResponse(questionnaire) : undefined;
+    });
   }, [questionnaire]);
 
+  useEffect(() => {
```

Comment:
I think this should work for pretty much every case we care about 👍 

If it proves to cause `onChange` to fire excessively, we can investigate using `usePrevious`